### PR TITLE
feat(RTLplay): add presence

### LINKED ISSUE #7282: RTLplay | www.rtlplay.be
### Website name

RTLplay

### Website URL

https://www.rtlplay.be/

### Website logo

https://images-fio.6play.fr/v2/images/3963687/raw?blur=0&fit=scale_crop&format=jpeg&height=630&interlace=1&quality=60&width=1200&hash=f92816700c781f4e57bdc55b48cef8d49fffecbe

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [X] It is region restricted

### Description

Available only in Belgium and Luxembourg.
It is the streaming service of one of the main media group in Belgium.
On the service you can watch for free all of the live streams of the channels
and also watch a lot of episodes of different tv series and replay of emissions.

Might be a good idea to detect when watching a livestream and when watching a replay or tv show.

### PR DESCRIPTION
**Pull request by pierrequiroul**
## Description 

Add presence for RTLplay. It's a streaming service where you can find all the programs live and in replay of RTL TVI, Club RTL, Plug RTL, Bel RTL and Radio Contact channels. Those channels are part of one of the main television group in Belgium.
List of assets and screenshots : https://imgur.com/a/iC5CIKJ

- Resolves #7282

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/R/RTLplay/presence.ts [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+		buttonWatchStream: "general.buttonWatchStream",
+		buttonWatchVideo: "general.buttonWatchVideo",
+		live: "general.live",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
+	Logo = "https://i.imgur.com/KNsI47l.png",
+	Animated = "https://imgur.com/uAqZdFg.gif",
+	RTLplay = "https://i.imgur.com/1f5rMxV.png",
```
Comment (DarkVillager) [inline websites/R/RTLplay/presence.ts:30]:
```suggestion
	RTLPlay = "https://i.imgur.com/1f5rMxV.png",
```

File: websites/R/RTLplay/presence.ts [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+	RTLTVi = "https://i.imgur.com/wnjbhCe.png",
+	RTLclub = "https://i.imgur.com/8FwZa7m.png",
+	RTLplug = "https://i.imgur.com/jfYLcJu.png",
```
Comment (DarkVillager) [inline websites/R/RTLplay/presence.ts:33]:
```suggestion
	RTLClub = "https://i.imgur.com/8FwZa7m.png",
	RTLPlug = "https://i.imgur.com/jfYLcJu.png",
```

File: websites/R/RTLplay/presence.ts [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+				channel: "RTL TVi",
+				type: ActivityType.Watching,
+				logo: Assets.RTLTVi,
+			};
+		}
+		case channel.includes("club"): {
+			return {
+				channel: "RTL club",
+				type: ActivityType.Watching,
+				logo: Assets.RTLclub,
```
Comment (DarkVillager) [inline websites/R/RTLplay/presence.ts:51]:
```suggestion
				logo: Assets.RTLClub,
```

File: websites/R/RTLplay/presence.ts [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+			};
+		}
+		case channel.includes("plug"): {
+			return {
+				channel: "RTL plug",
+				type: ActivityType.Watching,
+				logo: Assets.RTLplug,
```
Comment (DarkVillager) [inline websites/R/RTLplay/presence.ts:58]:
```suggestion
				logo: Assets.RTLPlug,
```

File: websites/R/RTLplay/presence.ts [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+				channel: "Radio Contact",
+				type: ActivityType.Listening,
+				logo: Assets.Contact,
+			};
+		}
+		default: {
+			return {
+				channel,
+				type: ActivityType.Watching,
+				logo: Assets.RTLplay,
```
Comment (DarkVillager) [inline websites/R/RTLplay/presence.ts:79]:
```suggestion
				logo: Assets.RTLPlay,
```

File: websites/R/RTLplay/presence.ts [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+			break;
+		}
+
+		/* Category (Theme) page 
+	Page Catégorie (Thématique)
+	(ex: https://www.rtlplay.be/rtl_play/divertissement-rtl-play-f_405)*/
+		case document.querySelector("div > header > div > div > h1") !== null ||
+			document.querySelector(
+				"div > header > div > div > nav > div.sc-nqi6nw-0.cNDqJj > h1"
+			) !== null: {
```
Comment (DarkVillager) [inline websites/R/RTLplay/presence.ts:299]: Try to shorten selectors. E.g.
```suggestion
			document.querySelector('[aria-current="page"]') !== null: {
```

File: websites/R/RTLplay/presence.ts [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+			break;
+		}
+	}
+
+	if ((presenceData.startTimestamp || presenceData.endTimestamp) && !time) {
+		delete presenceData.startTimestamp;
+		delete presenceData.endTimestamp;
+	}
+	if (presenceData.details === "") delete presenceData.details;
+	if (presenceData.state === "") delete presenceData.state;
```
Comment (DarkVillager) [inline websites/R/RTLplay/presence.ts:416]:
```suggestion
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/R/RTLplay/presence.ts (hunk 7 above)
Comment (pierrequiroul) [reply websites/R/RTLplay/presence.ts:416]: Does Premid automatically do that so those lines are not necessary ?
I added those lines to respect presence.ts rules
https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets (9th dot).

**Comment by DarkVillager** [comment]
Resolves #7282

File: websites/R/RTLplay/presence.ts [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+		const pdURL = await (
+			await fetch(`https://pd.premid.app/create/${url}`)
+		).text();
+		shortenedURLs[url] = pdURL;
+		return pdURL;
+	} catch (err) {
+		presence.error(err);
+		return url;
+	}
+}*/
```
Comment (theusaf) [inline websites/R/RTLplay/presence.ts:104]: Remove commented code

File: websites/R/RTLplay/presence.ts [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+		watchingAd: "youtube.ad",
+		watchingLive: "general.watchingLive",
+		watchingShow: "general.watchingShow",
+		buttonWatchStream: "general.buttonWatchStream",
+		buttonWatchVideo: "general.buttonWatchVideo",
+		live: "general.live",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
```
Comment (theusaf) [inline websites/R/RTLplay/presence.ts:27]:
```suggestion
const enum Assets {
```

File: websites/R/RTLplay/presence.ts [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+		return url;
+	}
+}*/
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Animated, // Default
+			largeImageText: "RTLplay",
+			type: ActivityType.Watching,
+		},
+		{ /*hostname,*/ href, pathname } = document.location,
```
Comment (theusaf) [inline websites/R/RTLplay/presence.ts:111]:
```suggestion
		{ href, pathname } = document.location,
```

File: websites/R/RTLplay/presence.ts [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+	(https://www.rtlplay.be/) */
+		case pathname === "/" || pathname === "/rtl_play": {
+			presenceData.details = (await strings).browsing;
+
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.smallImageText = (await strings).browsing;
+
+			if (time) presenceData.startTimestamp = browsingTimestamp;
+
+			break;
```
Comment (theusaf) [inline websites/R/RTLplay/presence.ts:130]: Remove extra line before `break`

File: websites/R/RTLplay/presence.ts [hunk 12]
```diff
@@ -0,0 +1,416 @@
...
+	The first one has a Play video button that the other one hasn't
+
+	(ex: https://www.rtlplay.be/chicago-fire-p_8947 and https://www.rtlplay.be/rtl_play/series-rtl-play-f_409) */
+		case exist(
+			"header > div > div > h1, div.sc-nqi6nw-0.cNDqJj > h1" // Movie name or Category name: Supports mobile and desktop layout
+		): {
+			presenceData.state = privacy
+				? ""
+				: document.querySelector(
+						"header > div > div > h1, div.sc-nqi6nw-0.cNDqJj > h1" // Movie name or Category name: Supports mobile and desktop layout
```
Comment (theusaf) [inline websites/R/RTLplay/presence.ts:269]: If possible, use selectors which use avoid computer-generated classes or selectors.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/R/RTLplay/presence.ts (hunk 12 above)
Comment (pierrequiroul) [reply websites/R/RTLplay/presence.ts:269]: Thanks for the feedback, I'll look into that !

**Comment by theusaf** [comment]
Closing for inactivity. You may open a new PR after resolving the issues mentioned in this one.

**Pull request closed without merging**